src/puppetlabs/puppetdb/queue.clj

```diff
@@ -30,15 +32,78 @@
     (catch Exception e
       (throw+ {:kind ::parse-error} e "Error parsing command"))))
 
+(defn sanitize-certname
+  "Replace any underscores and filename forbidden characters found in `certname`
+  with dashes."
+  [certname]
+  (let [forbidden-chars (conj constants/filename-forbidden-characters
+                              "_")]
+    (str/replace
+      certname
+      (re-pattern (str "("
+                       (->> forbidden-chars
```

Comment:
Since we do it in a couple places in this namespace alone, should we have a function for going from a collection of strings/characters to a regex (or perhaps regex string fragment to pass to re-pattern) that matches those strings/characters literally?